### PR TITLE
added procedure that allows to perform query on es and get back nodes from neo4j

### PR DESCRIPTION
**Pull request by alenegro81**

### CODE REVIEW

File: src/main/java/com/graphaware/module/es/proc/ElasticSearchProcedure.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        this.authUser = configuration.getAuthUser();
+        this.authPassword = configuration.getAuthPassword();
+        this.client = createClient();
+    }
+
+    public CallableProcedure.BasicProcedure query() {
+        return new CallableProcedure.BasicProcedure(procedureSignature(getProcedureName("query"))
+                .mode(ProcedureSignature.Mode.READ_WRITE)
+                .in(PARAMETER_NAME_INPUT, Neo4jTypes.NTMap)
+                .out(PARAMETER_NAME_OUTPUT, Neo4jTypes.NTNode).build()) {
```
Comment (ikwattro) [inline src/main/java/com/graphaware/module/es/proc/ElasticSearchProcedure.java]: Why not returning the `score` and `_source` ?

File: src/main/java/com/graphaware/module/es/proc/ElasticSearchProcedures.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Autowired
+    public ElasticSearchProcedures(GraphDatabaseService database, Procedures procedures) {
+        this.database = database;
+        this.procedures = procedures;
+    }
+
+    @PostConstruct
+    public void init() throws ProcedureException, KernelException {
+        ElasticSearchProcedure timeTreeProcedures = new ElasticSearchProcedure(database);
```
Comment (ikwattro) [inline src/main/java/com/graphaware/module/es/proc/ElasticSearchProcedures.java]: I would call the variable name something else than `timeTreeProcedures` :)

File: src/main/java/com/graphaware/module/es/proc/ElasticSearchProcedure.java (hunk 1 above)
Comment (ikwattro) [inline src/main/java/com/graphaware/module/es/proc/ElasticSearchProcedure.java]: Forget _source

**Comment by ikwattro** [comment]
In general 👍 

Just wondering, wouldn't it be more useful to return an object representing the `SearchResult` format we receive from Jest ? Where we would have a `List<Hit>` where Hit would contain the node and his score ?

The SearchResult could also contain totalHits and maxScore, this can be useful for calculating a new score further in the Cypher query.

File: src/main/java/com/graphaware/module/es/proc/ElasticSearchProcedures.java (hunk 2 above)
Comment (alenegro81) [inline src/main/java/com/graphaware/module/es/proc/ElasticSearchProcedures.java]: for sure :-)

**Comment by alenegro81** [comment]
I can return both of them without creating an object. I'll do it now

**Comment by alenegro81** [comment]
added score in the return values. 

Thanks @ikwattro for your suggestion

**Pull request merged**